Sprint 6 solution

ZotovSlava opened the pull request:
Доработана функциональность класса InMemoryHistoryManager в соответсвии заданием СПРИНТА №6. 

1)  История запросов неограничена по размеру.
2)  В истории нет дубликатов. Остается только последний запрос, а дублирующий удаляется.
3)  При удалении задачи из программы удаляется история ее просмотров. (При удалении Эпиков удаляются и его Сабтаски)

ZotovSlava committed:
Версия № 4:

1) Изменена логика работы с историей запросов класс InMemoryHistoryManager;
2) Изменен класс InMemoryHistoryManagerTest под новую логику.

ZotovSlava committed:
Версия № 4:

1) Изменена логика работы с историей запросов класс InMemoryHistoryManager;
2) Изменен класс InMemoryHistoryManagerTest под новую логику.

ZotovSlava committed:
Версия № 4.1:

1) Исправлены замечания GitHabTeat.

litvitnik commented on src/Node.java:
```diff
@@ -0,0 +1,11 @@
+public class Node<T> {
```
Класс Node должен быть объявлен в классе InMemoryHistoryManager и он должен быть статическим

litvitnik commented on src/InMemoryHistoryManager.java:
```diff
@@ -1,19 +1,113 @@
+    }
+
+
+    private static class MyLinkedList<T> {
```
Этот класс необходимо удалить, так как реализация указанных методов должна находиться в классе InMemoryHistoryManager

ZotovSlava committed:
Версия № 4.2:

1) Вложенный класс MyLinkedList удален. Его логика его работы реализована в классе InMemoryHistoryManager;

2) Класс Node стал статическим и вложенным в класс InMemoryHistoryManager.
